howto.md

```diff
@@ -280,51 +295,71 @@ Once in a while you will need to merge in changes from the main spec repo.
       
 If the merge conflicts are complex, you may want to have it reviewed. In that case:
 
-1. Merge upstream:
+1. Merge upstream with code review:
 
-   1. Run:
+   1. Make sure you are on the `master` branch
+
+   2. Pick a new `<<branch-name>>` for performing the merge and run:
       ```
-      git checkout -b  <<branch name>>
+      git checkout -b <<branch-name>>
       git fetch upstream
       git merge upstream/master
       ```
       
-   2. Resolve merge conflicts
+   3. Resolve merge conflicts
    
-   3. Run:
+   4. Run:
       ```
       git commit -m "Merge from upstream spec"
       git push
       ```
 
-2. Create a nicer diff:
+   5. Create a nicer diff:
 
-   The GitHub UI will display this as a list of all the changes added from the spec to this repo, which is not very easy to review. Instead, you can create a diff between this repo and the spec [merge base](https://git-scm.com/docs/git-merge-base).
+      The GitHub UI will display this as a list of all the changes added from the spec to this repo, which is not very easy to review. Instead, you can create a diff between this repo and the spec [merge base](https://git-scm.com/docs/git-merge-base).
    
-   1. Run:
-      ```
-      git diff upstream/master..origin/<<branch name>> > <<branch name>>.diff
-      ```
+      1. Run:
+         ```
+         git diff upstream/master..origin/<<branch-name>> > <<branch-name>>.diff
+         ```
       
-   2. Go to gist.github.com
+      2. Go to gist.github.com
    
-   3. Paste the contents of `<<branch name>>.diff` into the edit box
+      3. Paste the contents of `<<branch-name>>.diff` into the edit box
    
-   4. Name the file `<<branch name>>.diff`. This will make the diff have proper syntax highlighting.
+      4. Name the file `<<branch-name>>.diff`.
+         This will make the diff have proper syntax highlighting.
    
-   5. Click "Create public gist"
+      5. Click "Create public gist"
    
-   6. Copy the URL
+      6. Copy the URL
    
-3. Open a new pull request
+   6. Create a new pull request
 
-   1. Go to `https://github.com/WebAssembly/<<proposal>>/compare`
+      1. Go to `https://github.com/WebAssembly/<<proposal>>/compare`
    
-   2. Click the right branch button and choose `<<branch name>>`
+      2. Click the right branch button and choose `<<branch-name>>`
    
-   3. Click "Create pull request"
+      3. Click "Create pull request"
    
-   4. Fill out the details, and make sure to include the URL of your gist
+      4. Fill out the details, and make sure to include the URL of your gist
+
+   7. Once approved, commit
+
+      1. Do **not** use GitHub's "Squash and merge" option for committing upstream changes!
+         That will ruin the history and reiterate all merge conflicts the next time.
+
+      2. Instead, select "Merge pull request" in GitHub!
+
+      3. Alternatively, you can push manually:
+         ```
+         git checkout master
+         git rebase <<branch-name>>
```

Comment:
I don't think you want to use rebase here. Instead, you can do `git merge --ff-only <<branch-name>>`, which will only merge if it can be fast-forwarded, and won't create a merge commit.

Reply:
Changed.